Tasks/SonarQubePostTest/SonarQubePostTest.ps1

```diff
@@ -2,11 +2,19 @@ Write-Verbose "Starting SonarQube PostBuild Step"
 
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 
-. ./Common/SonarQubeHelpers/SonarQubeHelper.ps1
-. ./SonarQubePostTestImpl.ps1
-. ./SonarQubeBuildBreaker.ps1
-. ./CodeAnalysisFilePathComputation.ps1
+. $PSScriptRoot/Common/SonarQubeHelpers/SonarQubeHelper.ps1
+
+if (IsPrBuild -and (GetTaskContextVariable "DisableSQAnalysisOnPrBuilds" -eq "true")) 
+{
+	Write-Host "DisableSQAnalysisOnPrBuilds is set and this is a PR build - ignoring the analysis tasks"
+	return
+}
+
+. $PSScriptRoot/SonarQubePostTestImpl.ps1
+. $PSScriptRoot/SonarQubeBuildBreaker.ps1
+. $PSScriptRoot/CodeAnalysisFilePathComputation.ps1
 
 InvokeMSBuildRunnerPostTest
 UploadSummaryMdReport
+HandleCodeAnalysisReporting
```

Comment:
This statement is part of the original work to enable PR-CA and should appear on the left as well.